codec/types/interface_registry.go

```diff
@@ -0,0 +1,165 @@
+package types
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+// AnyUnpacker is an interface which allows safely unpacking types packed
```

Comment:
Is this source code that was copied from Google's implementation?

Reply:
No, only what's in any.go and any.pb.go.